src/openms/source/CHEMISTRY/SimpleTSGXLMS.cpp

```diff
@@ -404,77 +408,55 @@ namespace OpenMS
         mono_weight -= peptide[i].getMonoWeight(Residue::Internal);
         double pos(mono_weight / static_cast<double>(charge));
 
+        if (add_isotopes_ && max_isotope_ >= 2) // add second isotopic peak with fast method, if two or more peaks are asked for
+        {
+          addPeak_(spectrum, pos+(Constants::C13C12_MASSDIFF_U / static_cast<double>(charge)), charge);
+        }
         addPeak_(spectrum, pos, charge);
         if (add_losses_ && backward_losses.size() >= i+2)
         {
           addLosses_(spectrum, mono_weight, charge, backward_losses[i+1]);
         }
-
-        if (add_isotopes_ && max_isotope_ >= 2) // add second isotopic peak with fast method, if two or more peaks are asked for
-        {
-          pos += Constants::C13C12_MASSDIFF_U / static_cast<double>(charge);
-          addPeak_(spectrum, pos, charge);
-        }
       }
     }
     return;
   }
 
-
   // helper to add a single peak to a spectrum (simple fragmentation)
   void SimpleTSGXLMS::addPeak_(std::vector< SimplePeak >& spectrum, double pos, int charge) const
   {
-    if (pos < 0)
-    {
-      return;
-    }
-    SimpleTSGXLMS::SimplePeak p;
-    p.mz = pos;
-    p.charge = charge;
-    spectrum.push_back(p);
+    spectrum.resize(spectrum.size() + 1);
+    spectrum.back().mz = pos;
+    spectrum.back().charge = charge;
   }
 
   void SimpleTSGXLMS::addPrecursorPeaks_(std::vector< SimplePeak >& spectrum, double precursor_mass, int charge) const
   {
-    SimpleTSGXLMS::SimplePeak p;
-    p.charge = charge;
-
     // precursor peak
     double mono_pos = precursor_mass + (Constants::PROTON_MASS_U * static_cast<double>(charge));
-    p.mz = mono_pos / static_cast<double>(charge);
-    spectrum.push_back(p);
-
     if (add_isotopes_ && max_isotope_ >= 2) // add second isotopic peak with fast method, if two or more peaks are asked for
     {
-      double pos = mono_pos + (Constants::C13C12_MASSDIFF_U / static_cast<double>(charge));
-      p.mz = pos;
-      spectrum.push_back(p);
+      addPeak_(spectrum, (mono_pos + Constants::C13C12_MASSDIFF_U) / static_cast<double>(charge), charge);
```

Comment:
explicit casting to double is not required if one of the operands is a double already IMHO. Is there a warning otherwise?